src/FormattedText.tsx

```diff
@@ -0,0 +1,246 @@
+import * as React from 'react'
+import * as Marked from 'marked'
+import * as He from 'he'
+
+
+interface Props {
+    text: string,
+    format: string
+}
+
+interface State {
+}
+
+export class FormattedText extends React.Component<Props, State> {
+
+    shouldComponentUpdate(nextProps: Props, nextState: State, nextContext: any): boolean {
+        return this.props.text !== nextProps.text || this.props.format !== nextProps.format
+    }
+
+    render(): React.ReactElement<any> {
```

Comment:
I'm all for typing, but does adding the return type here (and in the rest of these functions) actually buy us anything?